### DISCUSSION TITLE
Create CONTRIBUTING.md

### DISCUSSION THREAD

**Discussion by joshmedeski** [LazyVim/LazyVim#1406] [Ideas]
I'm a big fan of LazyVim and recently contributed to the project. However, I wasn't sure the right steps to get it setup for testing on my machine.

I tried swapping out "lazyvim/lazyvim" in my config files, but got this:

```js
   Error  11:06:12 notify.error lazy.nvim Two plugins with the same name and different url:
{
  new = { "LazyVim/LazyVim",
    _ = {},
    cond = true,
    config = true,
    dir = "/Users/josh/.local/share/nvim/lazy/LazyVim",
    lazy = false,
    name = "LazyVim",
    priority = 10000,
    url = "https://github.com/LazyVim/LazyVim.git",
    version = "*"
  },
  old = { "joshmedeski/LazyVim",
    _ = {},
    dir = "/Users/josh/.local/share/nvim/lazy/LazyVim",
    import = "lazyvim.plugins",
    name = "LazyVim",
    url = "https://github.com/joshmedeski/LazyVim.git"
  }
}
```

I tried switching to `dev` and then got lost. I ended up ignoring these errors in order to finish my contribution but I think it would be helpful to create clear instructions for how people can fork, configure, and develop changes to LazyVim as well as any expectations or requirements you have for what you expect from contributors (conventional commits, one feature per PR, etc...)

I'm happy to help write up the document however I'm not sure about the technical steps required to have a fork of LazyVim setup on my local machine without any errors.

**Comment by dpetka2001** [comment]
I just take advantage of `NVIM_APPNAME` introduced in Neovim 0.9 onwards, because I don't like to introduce the various plugins I have installed during tinkering with Lazyvim contributing. So I just created a `~/.config/LazyVimDev` directory and followed the procedure to install the starter template in there. I also forked the Lazyvim repo and cloned it in `~/projects` and then in `/lua/config/lazy.lua` I just put the following towards the end of the file (or wherever) 
```lua
  dev = {
    path = "~/projects/",
    patterns = { "LazyVim" },
  },
```
With this addition Lazyvim is updated based on your local repo and you can create branches to work on features/bugs you want and see how Lazyvim behaves with the additions you make.

**Reply by dpetka2001** [reply]
You might want to read through #1129, which I opened in the past, which contains some info additional to my previous post you might or might not already know.
